kof-operator/internal/controller/clusterdeployment_kof_cluster_role.go

```diff
@@ -170,3 +180,80 @@ func (r *ClusterDeploymentReconciler) reconcileChildClusterRole(
 	)
 	return nil
 }
+
+func getCloud(clusterDeployment *kcmv1alpha1.ClusterDeployment) string {
+	cloud, _, _ := strings.Cut(clusterDeployment.Spec.Template, "-")
+	return cloud
+}
+
+func (r *ClusterDeploymentReconciler) discoverRegionalClusterDeploymentByLocation(
+	ctx context.Context,
+	childClusterDeployment *kcmv1alpha1.ClusterDeployment,
+	childClusterDeploymentConfig *ClusterDeploymentConfig,
+) (*kcmv1alpha1.ClusterDeployment, error) {
+	childCloud := getCloud(childClusterDeployment)
+
+	clusterDeploymentList := &kcmv1alpha1.ClusterDeploymentList{}
+	for {
```

Comment:
This looks overcomplicated, let's remove the Continue logic and add the selector option to the List method to get all Regional clusters by label at one request.